pkg/provider/gitlab/gitlab.go

```diff
@@ -214,26 +243,48 @@ func (g *Gitlab) GetSecret(ctx context.Context, ref esv1beta1.ExternalSecretData
 	// 	"masked": true,
 	// 	"environment_scope": "*"
 	// }
-
 	var vopts *gitlab.GetProjectVariableOptions
 	if g.environment != "" {
 		vopts = &gitlab.GetProjectVariableOptions{Filter: &gitlab.VariableFilter{EnvironmentScope: g.environment}}
 	}
-	data, _, err := g.client.GetVariable(g.projectID, ref.Key, vopts)
-	if err != nil {
+
+	data, resp, err := g.projectClient.GetVariable(g.projectID, ref.Key, vopts)
+	if resp.StatusCode >= 400 && err != nil {
 		return nil, err
 	}
 
+	var result []byte
+	result, err = extractVariable(ref, data.Value)
```

Comment:
We shouldn't ignore an error. Also, we should probably return early here before we even start the loop below.

Reply:
The error is conditionally ignored in case it is able to find the requested secret in the group variables.